pkg/operator/configobserver/apiserver/observe_tlssecurityprofile.go

```diff
@@ -24,33 +24,46 @@ type APIServerLister interface {
 	PreRunHasSynced() []cache.InformerSynced
 }
 
-// ObserveTLSSecurityProfile observes APIServer.Spec.TLSSecurityProfile field and sets
-// the ServingInfo.MinTLSVersion, ServingInfo.CipherSuites fields of observed config
-func ObserveTLSSecurityProfile(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
-	return innerTLSSecurityProfileObservations(genericListers, recorder, existingConfig, []string{"servingInfo", "minTLSVersion"}, []string{"servingInfo", "cipherSuites"})
+type tlsSecurityProfileObserver struct {
+	minTLSVersionPath []string
+	cipherSuitesPath  []string
 }
 
-// ObserveTLSSecurityProfileToArguments observes APIServer.Spec.TLSSecurityProfile field and sets
-// the tls-min-version and tls-cipher-suites fileds of observedConfig.apiServerArguments
-func ObserveTLSSecurityProfileToArguments(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
-	return innerTLSSecurityProfileObservations(genericListers, recorder, existingConfig, []string{"apiServerArguments", "tls-min-version"}, []string{"apiServerArguments", "tls-cipher-suites"})
+// NewTLSSecurityProfileObserver creates a new observer for the tls tlsSecurityProfile field,
+// it observes to the `servingInfo.minTLSVersion` and `servingInfo.cipherSuites`
+// or `apiServerArguments.tls-min-version` and `apiServerArguments.tls-cipher-suites` fields
+// depending on the useArguments argument
+func NewTLSSecurityProfileObserver(useArguments bool, prefix ...string) configobserver.ObserveConfigFunc {
+	var minTLSVersionPath []string
+	var cipherSuitesPath []string
+	if useArguments {
+		minTLSVersionPath = append(prefix, "apiServerArguments", "tls-min-version")
+		cipherSuitesPath = append(prefix, "apiServerArguments", "tls-cipher-suites")
+	} else {
+		minTLSVersionPath = append(prefix, "servingInfo", "minTLSVersion")
+		cipherSuitesPath = append(prefix, "servingInfo", "cipherSuites")
+	}
+
+	return (&tlsSecurityProfileObserver{
```

Comment:
nit: instead of creating a new type `tlsSecurityProfileObserver` we could return a function that would conform to `configobserver.ObserveConfigFunc` and would capture `minTLSVersionPath cipherSuitesPath`